editor.planx.uk/src/pages/FlowEditor/components/Flow/components/Portal.tsx

```diff
@@ -43,9 +43,13 @@ const ExternalPortal: React.FC<any> = React.memo(
       }),
     });
 
-    if (!data?.flows_by_pk) {
-      return null;
-    }
+    // If the flow referenced by an external portal has been deleted (eg !data),
+    //   still show a "Corrupted!" node so that editors have a visual cue to "delete".
+    //   The flow schema will still contain a node reference to this portal causing dataMerged to fail
+
+    // if (!data?.flows_by_pk) {
+    //   return null;
+    // }
```

Comment:
it felt most intuitive to me to add a comment here and comment out the seemingly-sensible faulty code (so the comment has context), rather than exclusively delete here - but open to other opinions here!